packages/gatsby/cache-dir/static-entry.js

```diff
@@ -199,17 +200,13 @@ export default (pagePath, callback) => {
     }
   }
 
-  const routerElement = createElement(
-    ServerLocation,
-    { url: `${__BASE_PATH__}${pagePath}` },
-    createElement(
-      Router,
-      {
-        id: `gatsby-focus-wrapper`,
-        baseuri: `${__BASE_PATH__}`,
-      },
-      createElement(RouteHandler, { path: `/*` })
-    )
+  const routerElement = (
+    <ServerLocation url={`${__BASE_PATH__}${pagePath}`}>
+      <Router id="gatsby-focus-wrapper" baseuri={__BASE_PATH__}>
+        <RouteHandler path="/*" />
+      </Router>
+      <RouteAnnouncer />
```

Comment:
This might not be ideal to rerender it for every route separately - might we could render it once and reuse string?

Also I'm not passing location here (it's only needed when this is in client), but is somewhat a red flag

Reply:
def red flag. This would crash on any update because `this.props.location.pathname` wouldn't exist

Reply:
`static-entry` is used for SSR only, so there never will be updates (at least in current state of things). I might just abstract the `<div>` with attributes and use it here. And `<RouteAnnouncer>` would make use of that specialized `<div>`